stack.c

```diff
@@ -5,55 +5,50 @@
 #include "stack.h"
 
 void push(Stack* stack, stackType data) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        printf("Stack overflow while push()");
+   if(stack->top == MAX_SIZE) {
+        printf("Stack overflow");
         return;
     }
-    newNode->data = data;
-    newNode->next = stack->top;
-    stack->top = newNode;
+    stack->top+=1;
+    stack->a[stack->top]=data;
 }
 
 stackType pop(Stack* stack) {
-    if(stack->top == NULL) {
-        printf("Stack underflow while pop\n");
-        return -1;
+    if(stack->top == -1) {
+        printf("Stack underflow\n");
+        return ;
     }
-    Node* next = stack->top->next;
-    stackType data = stack->top->data;
-    free(stack->top);
-    stack->top = next;
+    stackType data=stack->a[stack->top];
+    stack->top=stack->top-1;
     return data;
 }
 
 stackType peek(Stack* stack) {
-    if(stack->top == NULL) {
-        printf("Stack underflow while pop\n");
-        return -1;
+   if(stack->top == -1) {
+        printf("Stack is empty\n");
+        return ;
     }
-    stackType data = stack->top->data;
+    stackType data = stack->a[stack->top];
     return data;
 }
 
 int isEmpty(Stack* stack) {
-    return stack->top == NULL;
+    return stack->top == -1;
 }
 
 int isFull(Stack* stack) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        return 1;
-    }
-    free(newNode);
-    return 0;
+    return stack->top == MAX_SIZE;
 }
 
 void print(Stack* stack) {
-    Node* current = stack->top;
-    while(current != NULL) {
-        printf(TYPE_FMT" ", current->data);
-        current = current->next;
-    }
-    printf("\n");
-}
\ No newline at end of file
+   if(isEmpty(stack))
+    	return;
+    	
+    int i;
+    
+    for(i=0;i<=stack->top;i++){
```

Comment:
The original implementation printed the top of the stack first and then the rest. This one does the opposite. Not necessary to change.